jax/_src/pjit.py

```diff
@@ -957,16 +1072,25 @@ def _check_and_canonicalize_out_shardings(
 
 
 def _pjit_jaxpr(fun, out_shardings_thunk, out_layouts_thunk, in_type, debug_info,
-                device_or_backend_set, out_tree, result_paths):
+                device_or_backend_set, out_tree, result_paths, inline):
   jaxpr, final_consts, out_type = _create_pjit_jaxpr(
-      fun, in_type, debug_info, result_paths)
+      fun, in_type, debug_info, result_paths, IgnoreKey(inline))
   canonicalized_out_shardings_flat, out_layouts_flat = _check_and_canonicalize_out_shardings(
       out_shardings_thunk, out_layouts_thunk, out_tree, tuple(out_type),
       jaxpr.jaxpr.debug_info, device_or_backend_set)
   # lu.cache needs to be able to create weakrefs to outputs, so we can't return a plain tuple
   return jaxpr, final_consts, canonicalized_out_shardings_flat, out_layouts_flat
 
 
+@dataclasses.dataclass(frozen=True)
+class IgnoreKey:
+  val: Any
+  def __hash__(self):
+    return hash(self.__class__)
+  def __eq__(self, other):
+    return type(other) is IgnoreKey  # ignore self.val!
```

Comment:
```suggestion
    return isinstance(other, IgnoreKey)  # ignore self.val!
```

Reply:
`type` is preferred according to the JAX Style Guide...

Reply:
Yeah but mypy and pytype doesn't like it ;)

Reply:
Done!